server/app/controllers/api/ApiDocsController.java

```diff
@@ -34,32 +36,49 @@ public ApiDocsController(
   public Result index(Http.Request request) {
     Optional<String> firstProgramSlug = programService.getAllProgramSlugs().stream().findFirst();
     return firstProgramSlug
-        .map(slug -> redirect(routes.ApiDocsController.docsForSlug(slug)))
+        .map(slug -> redirect(routes.ApiDocsController.activeDocsForSlug(slug)))
         .orElse(
             notFound(
-                "No active programs found. Please create and publish a program before accessing"
+                "No programs found. Please create and publish a program before accessing"
                     + " API docs."));
   }
 
-  public Result docsForSlug(Http.Request request, String programSlug) {
+  public Result activeDocsForSlug(Http.Request request, String selectedProgramSlug) {
+    return docsForSlug(request, selectedProgramSlug, /* useActiveVersion= */ true);
+  }
+
+  public Result draftDocsForSlug(Http.Request request, String selectedProgramSlug) {
+    return docsForSlug(request, selectedProgramSlug, /* useActiveVersion= */ false);
+  }
+
+  private Result docsForSlug(
+      Http.Request request, String selectedProgramSlug, boolean useActiveVersion) {
     if (!settingsManifest.getApiGeneratedDocsEnabled()) {
       return notFound("API Docs are not enabled.");
     }
 
     ImmutableSet<String> allProgramSlugs = programService.getAllProgramSlugs();
+    Optional<ProgramDefinition> programDefinition =
+        getProgramDefinition(selectedProgramSlug, useActiveVersion);
+
+    return ok(docsView.render(request, selectedProgramSlug, programDefinition, allProgramSlugs));
+  }
+
+  private Optional<ProgramDefinition> getProgramDefinition(
+      String programSlug, boolean useActiveVersion) {
+
+    Function<String, CompletionStage<ProgramDefinition>> getProgramDefFn =
+        useActiveVersion
+            ? programService::getActiveProgramDefinitionAsync
+            : programService::getDraftProgramDefinitionAsync;
 
-    ProgramDefinition programDefinition;
     try {
-      programDefinition =
-          programService.getActiveProgramDefinitionAsync(programSlug).toCompletableFuture().join();
+      ProgramDefinition programDefinition =
+          getProgramDefFn.apply(programSlug).toCompletableFuture().join();
+      return Optional.of(programDefinition);
+
     } catch (Exception e) {
```

Comment:
**suggestion (blocking):** I know you didn't add this line, but we should **never** catch `Exception` like this, please update to a more specific exception type.

Reply:
Updated to RuntimeException. I'll leave this comment unresolved until we figure out the best approach to throwing in the other method, since this might change.

Reply:
https://github.com/civiform/civiform/pull/5315 😄

Reply:
Thanks!